common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -306,17 +325,30 @@ func (s *InterleavedWeightedRoundRobinScheduler[T, K]) doDispatchTasksWithWeight
 	channels WeightedChannels[T],
 ) {
 	numTasks := int64(0)
+	needChannelCleanup := false
 LoopDispatch:
 	for _, channel := range channels {
 		select {
 		case task := <-channel.Chan():
+			channel.UpdateLastActiveTime(s.ts.Now())
 			s.fifoScheduler.Submit(task)
 			numTasks++
 		default:
+			if s.options.InactiveChannelDeletionDelay != nil &&
+				s.ts.Now().Sub(channel.lastActiveTime) > s.options.InactiveChannelDeletionDelay() {
+
+				needChannelCleanup = true
+			}
```

Comment:
I am thinking if a separate goroutine that periodically cleans up channel is better. Then we don't have this overhead on each dispatch. Also the check can be done on the non-flattened version.

Reply:
Hmm. I also considered that approach. My reasoning was that we are iterating through this list anyway, and it would not add much overhead to just check a condition here. The actual cleanup will not happen that frequently. 

But it also makes sense to keep dispatch loop simple as you said.

Reply:
Added a cleanup loop which will do a periodic cleanup.